codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/protocols/eventstream/EventStreamBaseRequirements.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.protocols.eventstream
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
+import software.amazon.smithy.rust.codegen.client.smithy.customize.CombinedClientCodegenDecorator
+import software.amazon.smithy.rust.codegen.client.testutil.clientTestRustSettings
+import software.amazon.smithy.rust.codegen.client.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
+import software.amazon.smithy.rust.codegen.core.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.core.smithy.generators.BuilderGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.implBlock
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestRequirements
+
+abstract class EventStreamBaseRequirements : EventStreamTestRequirements<ClientCodegenContext> {
```

Comment:
Should we agree on a naming strategy once and for all on classes that have equivalents across `codegen-client` and `codegen-server`? _Most_ (but not all) of the classes in `codegen-server` are prefixed with `Server`, while few of the classes in `codegen-client` are prefixed with `Client`. Likewise, some classes from `codegen-core` that we inherit from the other modules are prefixed with `Core`.

My preference would be to not prefix things with `Core`, but have everything prefixed with `Client` and `Server` that comes from `codegen-client` and `codegen-server`, respectively. While redundant (the information is encoded in the classpath), I like that one can tell at a glance the purpose of things. This would only apply to classes that have equivalents across the subprojects (e.g. we have `ServerRestJson` in `codegen-server` and `RestJson` in `codegen-client`, so the latter should be renamed to `ClientRestJson`, but there's no need to prefix `ConstraintViolationSymbolProvider` with `Server`, since constraint traits are ignored in the client).

We could even attempt writing a lint that enforced the naming strategy in CI. What do you think?

Reply:
I'm fine with your proposed naming strategy. It definitely makes it easier to pull up the correct class in IntelliJ. I'll rename these ones.

Reply:
Fixed in 5c73370f087e40863b0960118bfb49e07325d926.